### PR TITLE
Enhance Find feature

### LINKED ISSUE #27: Implement additional find features
like find will match other components of the ride, like matching the location of the ride or the person operating the ride e.g
find <a> building 4
Results:	
BattleStar Galactica

### PR DESCRIPTION
**Pull request by keenlok**
Allow find to find other components of the rides through the use of the prefixes.
For now, only can search using address prefix and tag prefix

E.g _"find a/10th street "_ or _"find t/friends"_

Resolves #27

### CODE REVIEW

**Comment by keenlok** [comment]
Ahh I also added a test in the previous commit

**Comment by keenlok** [comment]
The hasAddress was used not to check the address field of the ride but whether the user included an address in their input. I'm not using it anymore so I can remove it now.

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 1]
```diff
@@ -26,11 +29,14 @@
...
      */
     public FindCommand parse(String args) throws ParseException {
-        ArgumentMultimap argMutlimap = ArgumentTokenizer.tokenize(args, PREFIX_ADDRESS);
-        Optional<Address> address = Optional.empty();
-        if (argMutlimap.getValue(PREFIX_ADDRESS).isPresent()) {
-            address = Optional.of(ParserUtil.parseAddress(argMutlimap.getValue(PREFIX_ADDRESS).get()));
-        }
+        ArgumentMultimap argMutlimap = ArgumentTokenizer.tokenize(args, PREFIX_ADDRESS, PREFIX_TAG);
+        Optional<Address> address =
+                !argMutlimap.getValue(PREFIX_ADDRESS).isPresent() ? Optional.empty()
```
Comment (YuPeiHenry) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java]: Is if/else preferred over ternary operators? I'm not sure, need to check on this.

File: src/test/java/systemtests/FindCommandSystemTest.java [hunk 2]
```diff
@@ -90,51 +93,64 @@ public void find() {
...
+        ModelHelper.setFilteredList(expectedModel);
         assertCommandSuccess(command, expectedModel);
-        // assertSelectedCardUnchanged();
+        assertSelectedCardUnchanged();
+
+        /* Case: find address of ride in thane park -> 3 rides found */
+        command = FindCommand.COMMAND_WORD + " " + PREFIX_ADDRESS + DUMBO.getAddress().value;
+        ModelHelper.setFilteredList(expectedModel, CASTLE, DUMBO, GALAXY);
+        assertCommandSuccess(command, expectedModel);
+        assertSelectedCardUnchanged();
```
[the remark this answers is not in the document]
Comment (YuPeiHenry) [reply src/test/java/systemtests/FindCommandSystemTest.java]: Sorry didn't see this.

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java (hunk 1 above)
Comment (YuPeiHenry) [reply src/main/java/seedu/address/logic/parser/FindCommandParser.java]: Please follow the formats:
alpha = (aLongBooleanExpression) ? beta : gamma;
alpha = (aLongBooleanExpression)
        ? beta
        : gamma;

**Pull request merged**